### DISCUSSION TITLE
Why is my tidyplot in the RStudio Viewer pane so small?

### DISCUSSION THREAD

**Discussion by jbengler** [jbengler/tidyplots#26] [Ask a question] [answered]
When generating a tidyplot, the preview in the RStudioViewer pane is too small.

**Comment by jbengler** [comment, marked as answer]
Unlike ggplot2, tidyplots uses **absolute sizes** in millimeter (mm) for plot area size. The plot area is the area that is enclosed by the two axes. This means that ticks, titles, caption, and legend are not counting towards the plot area size.

The default plot area size in tidyplots is 50 mm by 50 mm. This can be hard to see depending on your setup. There are several ways to address this.

1. In RStudio you can increase the zoom to enlarge the plot. `View` > `Zoom In`. Or alternatively `Ctrl` + `+`. However, this will also increase the font size in RStudio.

2. You can increase the absolute size of the tidyplot by providing an alternative `width` and `height` in mm either to the `tidyplot()` or `adjust_size()` function.

```{r eval=FALSE}
# using the tidyplot() function
study %>%
  tidyplot(x = treatment, y = score, color = treatment, width = 80, height = 80) %>%
  add_mean_dash() %>%
  add_sem_errorbar() %>%
  add_data_points_beeswarm()

# using the adjust_size() function
study %>%
  tidyplot(x = treatment, y = score, color = treatment) %>%
  add_mean_dash() %>%
  add_sem_errorbar() %>%
  add_data_points_beeswarm() %>% 
  adjust_size(80, 80)
```

3. You can let the tidyplot fill all available space (ggplot2 default). However with this solution the proportions of font size, line width, and plot size area will become arbitrary.

```{r eval=FALSE}
# using the tidyplot() function
study %>%
  tidyplot(x = treatment, y = score, color = treatment, width = NA, height = NA) %>%
  add_mean_dash() %>%
  add_sem_errorbar() %>%
  add_data_points_beeswarm()

# using the adjust_size() function
study %>%
  tidyplot(x = treatment, y = score, color = treatment) %>%
  add_mean_dash() %>%
  add_sem_errorbar() %>%
  add_data_points_beeswarm() %>% 
  adjust_size(NA, NA)
```

**Reply by briatte** [reply]
This solution does not seem to work with `split_plot`.

```r
study %>%
    tidyplot(x = treatment, y = score, color = treatment) %>%
    add_mean_dash() %>%
    add_sem_errorbar() %>%
    add_data_points_beeswarm() %>% 
    adjust_size(NA, NA) %>% 
    split_plot(by = sex)
```

Neither does adjusting size by passing different values than `NA`.

**Reply by jbengler** [reply]
Hi @briatte 

This is because `split_plot()` has its own `widths` and `heights` arguments.

Try this:

```
library(tidyplots)

study %>%
    tidyplot(x = treatment, y = score, color = treatment) %>%
    add_mean_dash() %>%
    add_sem_errorbar() %>%
    add_data_points_beeswarm() %>% 
    split_plot(by = sex)

study %>%
    tidyplot(x = treatment, y = score, color = treatment) %>%
    add_mean_dash() %>%
    add_sem_errorbar() %>%
    add_data_points_beeswarm() %>% 
    split_plot(by = sex, widths = NA, heights = NA)
```

Best
Jan

**Reply by briatte** [reply]
Alright, thanks! That does work.

**Comment by yudalang3** [comment]
This is a very big issue for data analysis #31 also metioned.
So, I really suggest author the state such a "A strange design" to the obvious location (for example, the index.html of the website, the get start on the web site and the github repo. main face)

**Comment by hasibagen** [comment]
Thank you for your efforts on the package. I understand that this might seem like a small issue, but it significantly affects both the plot results and the overall user experience.

I’ve tried all the methods mentioned so far, but unfortunately, none of them have worked in terms of getting the plot to display the correct size and label scale.

I can sense that the goal of this package is to simplify the plotting code and logic, and I truly appreciate that. However, if adjusting the size and labels requires a significant amount of time and effort, I might just revert to using ggplot2 directly to get the desired results.

Thanks again for your time and attention to this!

Best regards,
Jade

**Reply by jbengler** [reply]
Hi @hasibagen 

could you provide me with some more informations to better understand the problem?

Ideal would be your data and code and a detailed description of the desired output (PDF file, PNG file, RMarkdown document, shiny app, ...?)

Best
Broder
